platform/include/platform/mbed_retarget.h

```diff
@@ -37,7 +37,7 @@
 #endif
 #include <errno.h>
 
-#if !defined(__ARMCC_VERSION)
+#if !defined(__ARMCC_VERSION) && defined(__GNUC__)
```

Comment:
is this addition needed?

Reply:
https://github.com/ARMmbed/mbed-os/pull/14751#issuecomment-882467161